### PR TITLE
[xbmc] fix credentials in an URL in case of VFS add-on.

### PR DESCRIPTION
**Pull request by afedchin**
see title

## Description
currently we hide credentials for smb urls and store them into PasswordManager. Internal smb implementation knows about this and properly gets credentials from the manager and operates with full url but a VFS add-on can't do this. To fix this I've move this logic to lower level to CDirectory and CFile. This allows smb VFS add-on work properly. Also PasswordManager supports nfs/sftp now.

### CODE REVIEW

File: xbmc/PasswordManager.cpp [hunk 1]
```diff
@@ -105,6 +105,16 @@ void CPasswordManager::SaveAuthenticatedURL(const CURL &url, bool saveToProfile)
   m_temporaryCache[GetServerLookup(path)] = authenticatedPath;
 }
 
+bool CPasswordManager::IsURLSupported(const CURL &url)
+{
+  if ( url.IsProtocol("smb")
```
Comment (notspiff) [inline xbmc/PasswordManager.cpp]: style suggestion: return url.IsProtocol("smb") || ..

File: xbmc/PasswordManager.h [hunk 2]
```diff
@@ -79,6 +79,17 @@ class CPasswordManager
    */
   void SaveAuthenticatedURL(const CURL &url, bool saveToProfile = true);
 
+  /*!
+   \brief Save an authenticated URL.
+
+   This routine chacks that an URL is supported by the manager
```
Comment (notspiff) [inline xbmc/PasswordManager.h]: chEcks

**Review by notspiff — CHANGES_REQUESTED** [review]
Since this has been lifted up the stack, the callback should be removed from vfs add-on interface as well.

**Comment by afedchin** [comment]
> Since this has been lifted up the stack, the callback should be removed from vfs add-on interface as well.

I think I didn't get you right. The callback is required in case if credentials is still wrong. VFS addon may request enter credentials, and in next time VFS will be called with new credentials.

**Comment by notspiff** [comment]
ah, okay. i thought the query was automatic in the filesystem now.

**Review by notspiff — APPROVED** [review]
Then it's all good.

**Comment by afedchin** [comment]
behavior still same. filesystem can't know that credentials are wrong. VFS addon decides in case as an example connection established but refused by wrong login see https://github.com/afedchin/vfs.smb2/blob/master/src/SMBFile.cpp#L206

File: xbmc/PasswordManager.cpp (hunk 1 above)
Comment (afedchin) [reply xbmc/PasswordManager.cpp]: current style allows see a result in debugging.

**Comment by afedchin** [comment]
jenkins build this please

**Pull request merged**